src/ai/backend/client/func/user.py

```diff
@@ -242,6 +251,39 @@ async def detail_by_uuid(
         )
         return data["user_from_uuid"]
 
+    @api_function
+    @classmethod
+    async def detail_by_email(
+        cls,
+        email: str = None,
+        fields: Sequence[FieldSpec] = _user_info_fields,
+    ) -> Sequence[dict]:
+        """
+        Fetch basic information of a user by user's email. If email is not specified,
+        requester's information will be returned.
+
+        :param email: email of the user to fetch.
+        :param fields: Additional per-user query fields to fetch.
+        """
+        if email is None:
+            query = textwrap.dedent("""\
+                query {
+                    user {$fields}
+                }
+            """)
+        else:
+            query = textwrap.dedent("""\
+                query($email: String!) {
+                    user_from_email(email: $email) {$fields}
+                }
+            """)
+        query = query.replace("$fields", " ".join(f.field_ref for f in fields))
+        print("query{0}", query)
+        variables = {"email": email}
+        data = await api_session.get().Admin._query(query, variables if email is not None else None)
+        print("data from func", data["user_from_email"])
```

Comment:
The functional APIs in the client should avoid using direct stdout outputs. Please remove debugging-purpose print() calls and/or use the standard logging so that the caller can control the logs.